src/config/config.ts

```diff
@@ -0,0 +1,351 @@
+import convict from "convict"
+
+convict.addFormat({
+  name: "required-string",
+  validate: (val: any) => {
+    if (!val) throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "string") throw new Error("value must be a string")
+  },
+})
+
+convict.addFormat({
+  name: "required-positive-number",
+  validate: (val: any) => {
+    if (!val) throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "number") throw new Error("value must be a number")
+  },
+  coerce: (val: string) => {
+    const coercedVal = Number(val)
+    if (coercedVal < 0) {
+      throw new Error("value must be a positive number")
+    }
+    return coercedVal
+  },
+})
+
+convict.addFormat({
+  name: "required-boolean",
+  validate: (val: any) => {
+    if (val === null || val === undefined)
+      throw new Error("value cannot be empty, null or undefined")
+    if (typeof val !== "boolean") throw new Error("value must be a boolean")
+  },
+  coerce: (val: string) => String(val).toLowerCase() === "true",
+})
+
+// Define a schema
+const config = convict({
+  env: {
+    doc: "The application environment.",
+    env: "NODE_ENV",
+    format: ["LOCAL_DEV", "DEV", "test", "prod"],
+    default: "LOCAL_DEV",
+  },
+  port: {
+    doc: "The port to bind.",
+    env: "PORT",
+    format: "required-positive-number",
+    default: 8081,
+  },
+  gitGuardian: {
+    doc: "API Key for GitGuardian pre-commit hooks",
+    env: "GITGUARDIAN_API_KEY",
+    sensitive: true,
+    format: String,
+    default: "",
+  },
+  cloudmersiveKey: {
+    doc: "API Key for Cloudmersive scanning",
+    env: "CLOUDMERSIVE_API_KEY",
+    sensitive: true,
+    format: "required-string",
+    default: null,
+  },
+  app: {
+    frontendUrl: {
+      doc: "URL of the frontend application",
+      env: "FRONTEND_URL",
+      format: "required-string",
+      default: null,
+    },
+  },
+  mutexTableName: {
+    doc: "Name of the DynamoDB table used for mutexes",
+    env: "MUTEX_TABLE_NAME",
+    format: "required-string",
+    default: "isomer-mutexes",
+  },
+  sites: {
+    pageCount: {
+      doc: "Number of pages of repos to retrieve from GitHub API",
+      env: "ISOMERPAGES_REPO_PAGE_COUNT",
+      format: "required-positive-number",
+      default: 10,
+    },
+    localSiteAccessToken: {
+      doc: "Access token for local site",
+      env: "LOCAL_SITE_ACCESS_TOKEN",
+      sensitive: true,
+      format: "required-string",
+      default: null,
+    },
+  },
+  auth: {
+    cookieDomain: {
+      doc: "Domain to set for auth cookie",
+      env: "COOKIE_DOMAIN",
+      format: ["localhost", "cms.isomer.gov.sg", "isomer.gov.sg"],
```

Comment:
could i ask if this also includes the staging cookie domain?

Reply:
I checked aws console for staging EB and i included that as well. let me know if there should be others i should add